File: mobile/ios/FangornSentinel/Models/Alert.swift
import Foundation

struct Alert: Identifiable, Codable {
    let id: Int
    let title: String
    let message: String?
    let severity: Severity
    let status: Status
    let firedAt: Date
    let acknowledgedAt: Date?
    let source: String?

    enum Severity: String, Codable {
        case critical
        case warning
        case info

        var color: String {
            switch self {
            case .critical: return "red"
            case .warning: return "orange"
            case .info: return "blue"
            }
        }
    }

    enum Status: String, Codable {
        case firing
        case acknowledged
        case resolved
    }

    init(id: Int, title: String, message: String?, severity: Severity, status: Status, firedAt: Date, acknowledgedAt: Date? = nil, source: String? = nil) {
        self.id = id
        self.title = title
        self.message = message
        self.severity = severity
        self.status = status
        self.firedAt = firedAt
        self.acknowledgedAt = acknowledgedAt
        self.source = source
    }
}

File: mobile/android/app/src/main/java/com/notifd/fangornsentinel/models/Alert.kt
package com.notifd.fangornsentinel.models

import androidx.compose.ui.graphics.Color
import java.time.Instant

enum class AlertSeverity {
    CRITICAL,
    WARNING,
    INFO;

    fun toColor(): Color {
        return when (this) {
            CRITICAL -> Color.Red
            WARNING -> Color(0xFFFF9800) // Orange
            INFO -> Color.Blue
        }
    }
}

enum class AlertStatus {
    FIRING,
    ACKNOWLEDGED,
    RESOLVED
}

data class Alert(
    val id: String,
    val title: String,
    val message: String? = null,
    val severity: AlertSeverity,
    val status: AlertStatus,
    val source: String? = null,
    val sourceId: String? = null,
    val labels: Map<String, String> = emptyMap(),
    val annotations: Map<String, String> = emptyMap(),
    val firedAt: Instant,
    val acknowledgedAt: Instant? = null,
    val resolvedAt: Instant? = null,
    val assignedToId: String? = null
)
